lib/controllers/saml-redirect.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var stormpath = require('stormpath');
+
+/**
+ * This controller initiates a SAML login process, allowing the user to register
+ * via a registered SAML provider.
+ *
+ * When the user logs in through the SAML provider, they will be redirected back
+ * (to the SAML verification URL).
+ *
+ * @method
+ *
+ * @param {Object} req - The http request.
+ * @param {Object} res - The http response.
+ */
+module.exports = function (req, res) {
+  var application = req.app.get('stormpathApplication');
+  var builder = new stormpath.SamlIdpUrlBuilder(application);
+  var config = req.app.get('stormpathConfig');
+  var cbUri = req.protocol + '://' + req.get('host') + config.web.saml.verifyUri;
```

Comment:
I'm using a similar approach to support SAML login. I'd recommend support a `host` config option here instead of assuming the `req.host`. Running this in a Docker container behind an nginx proxy winds up with the `host` being reported as the Docker service name instead of the public URI.